### ISSUE TITLE
nmap 7.80 fails ncat addrset tests on ppc64be

### ISSUE THREAD

**Issue by mepholic** [nmap/nmap#1739]
Hello, I am the maintainer of the nmap package for Adélie Linux. During a recent attempted version bump from version 7.70 to 7.80, nmap began failing 3 of its tests in the addrset testsuite on ppc64be (big-endian). The testsuite appears to pass on x86_64. I suspect there may be an endian issue at play here given the highly bitwise nature of the code changes that are suspected to be at play here, but I'm not quite sure.

I've enabled some additional debugging in `ncat/test/addrset.c` by adding `nbase_set_log(loguser, logdebug);` to the top of the `main()` function. Below is the relevant output from the testsuite:
```
NCAT DEBUG: Add IP 192.168.0.0/24 to addrset (trie).
FAIL "192.168.0.5 192.168.0.90 192.168.1.5 1.2.3.4" !=
     "192.168.0.5 192.168.0.90".
NCAT DEBUG: Add IP 1:2::3/120 to addrset (trie).
FAIL "1:2::3 1:2::0 1:2::ff 1:2::1ff" !=
     "1:2::3 1:2::0 1:2::ff".
NCAT DEBUG: Add IPv4 45.33.32.156/30 to addrset (trie).
NCAT DEBUG: Add IPv6 2600:3c01::f03c:91ff:fe18:bb2f/30 to addrset (trie).
FAIL "1:2::3:4 1.2.3.4 scanme.nmap.org" !=
     "scanme.nmap.org".
```

In all these cases, the `addrset` program is returning all of the inputs specified, instead of just the items that match the mask. I found that these tests DO work properly when I pass in `localhost`. For example:
```
$ ./addrset "192.168.0.0/24"
NCAT DEBUG: Add IP 192.168.0.0/24 to addrset (trie).
192.168.0.5   <- stdin
192.168.0.5   <- stdout
192.168.0.90  <- stdin
192.168.0.90  <- stdout
192.168.1.5   <- stdin
192.168.1.5   <- stdout
1.2.3.4       <- stdin
1.2.3.4       <- stdout
localhost     <- stdin
^C
```

After fairly extensive testing, I've determined that these specific failures on ppc64be were introduced at commit 625884e7dcafae307fa563bc39deba4521d1cdbf

I've taken a look at some of the relevant code, but I haven't been able to determine the cause of the issue. The previous commit in the tree also extensively modifies the operation of many addrset functions, so there may be related changes in that commit as well.

Please let me know if there's anything that I can do to assist in fixing this issue. I can test code for you on hardware that reproduces this issue, and we may even be able to supply a virtual machine for testing if necessary. We have only attempted building on x86_64 and ppc64be so far, but we can also test on i586, ppc32be, armv7, and aarch64 if you think that would help in diagnosing the problem.

I've attached the full testsuite logs for completeness.
[nmap-7.80-check.log](https://github.com/nmap/nmap/files/3608096/nmap-7.80-check.log)

**Comment by nnposter** [comment]
This issue appears to be a duplicate of #1717. A fix has been committed as r37726 (9e8852a7c7dbc2f60fa1b9846ab684ec27db8b89).

Please give it a try and report back.

**Comment by mepholic** [comment]
This is indeed the fix! Sorry I didn't check outstanding issues first. I was working on this issue right after the 7.80 release and got caught up with $dayjob crap. I checked outstanding issues a few weeks ago and did not see #1717 

This fix has been applied to nmap in Adélie's package tree. Thank you for the help!
